subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/CleanupStaleOutputsExecuter.java

```diff
@@ -66,6 +68,7 @@ public void execute(final TaskInternal task, TaskStateInternal state, TaskExecut
             }
         }
         if (!filesToDelete.isEmpty()) {
+            taskOutputsGenerationListener.beforeTaskOutputsGenerated();
```

Comment:
Should we rename the method to `beforeTaskOutputsChanged()`?

Reply:
Renamed to `beforeTaskOutputChanged()`.